src/main/java/com/splunk/kafka/connect/SplunkSinkConnector.java

```diff
@@ -66,4 +80,105 @@ public String version() {
     public ConfigDef config() {
         return SplunkSinkConnectorConfig.conf();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Config config = super.validate(connectorConfigs);
+        if (config.configValues().stream().anyMatch(cv -> !cv.errorMessages().isEmpty())) {
+            return config;
+        }
+
+        Map<String, ConfigValue> configValues =
+            config.configValues()
+                .stream()
+                .collect(Collectors.toMap(
+                    ConfigValue::name,
+                    Function.identity()));
+
+        SplunkSinkConnectorConfig connectorConfig;
+        try {
+            connectorConfig = new SplunkSinkConnectorConfig(connectorConfigs);
+        } catch (Exception e) {
+            log.error("Validating configuration caught an exception", e);
+            return config;
+        }
+
+        validateAccess(connectorConfig, configValues);
+
+        return config;
+    }
+
+    /**
+     * We validate access by posting an empty payload to the Splunk endpoint.
+     *
+     * For a valid endpoint and a valid token, this returns a HTTP 400 Bad Request with the
+     * payload: {"text":"No data","code":5}
+     * For a valid endpoint and an invalid token, this returns a HTTP 403 Forbidden with the
+     * payload: {"text":"Invalid token","code":4}
+     * For an invalid hostname and other errors, the Java UnknownHostException and other similar
+     * Exceptions are thrown.
+     *
+     * @param connectorConfig The connector configuration
+     * @param configValues The configuration ConfigValues
+     */
+    private void validateAccess(SplunkSinkConnectorConfig connectorConfig, Map<String, ConfigValue> configValues) {
+        HecConfig config = connectorConfig.getHecConfig();
+
+        try (CloseableHttpClient httpClient = createHttpClient(config)) {
+            List<String> uris = config.getUris();
+            final String hecToken = config.getToken();
+
+            CountDownLatch latch = new CountDownLatch(config.getUris().size());
+            Map<String, String> validationFailedIndexers = new LinkedHashMap<>();
+
+            for (String uri : uris) {
+                log.info("Validating " + uri);
+                HttpPost request = new HttpPost(uri + "/services/collector");
+                request.setEntity(new StringEntity(""));
+
+                request.addHeader(HttpHeaders.AUTHORIZATION, String.format("Splunk %s", hecToken));
```

Comment:
```suggestion
                request.addHeader(HttpHeaders.AUTHORIZATION, String.format("Splunk %s", config.getToken()));
```
optional nit: Better to not load the secret into a variable exposing it inadvertently in logs unknowingly.

Reply:
Sure, done!